### PR TITLE
mobile app fixes

### PR DESCRIPTION
**Pull request by randyhoulahan**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -114,4 +118,23 @@ function setCustomCacheControl(res, path) {
...
 function send404(req, res) { 
   res.status(404).send(); 
+}
+
+function whiteListIframeUrls(req, res, next){
+    // short circuit
+    const [ appUrlRegEx, apiUrlRegEx ] = [ /^\/app\//, /^\/api\// ]
+
+    if(appUrlRegEx.test(req.url) || apiUrlRegEx.test(req.url)) return next();
+
```
Comment (sbilodeau) [inline server.js:128]: no needed
```suggestion
```

File: server.js [hunk 2]
```diff
@@ -114,4 +118,23 @@ function setCustomCacheControl(res, path) {
...
+    // white list
+    const iframeAllowedUrls = [
+        /^\/conferences\/(.*)\/schedules\?(viewOnly\=true|viewOnly)/,
+        /^\/conferences\/(.*)\/(.*)\/documents\?(viewOnly\=true|viewOnly)/
```
Comment (sbilodeau) [inline server.js:132]: use catchall 
```suggestion
        /^\/conferences\/(.*)\/schedules\?viewOnly.*/,
        /^\/conferences\/(.*)\/(.*)\/documents\?viewOnly.*/
```

**Review by sbilodeau — CHANGES_REQUESTED** [review]
Droppped some duplicate comde  + simplify rules (easier to read)

File: server.js [hunk 3]
```diff
@@ -114,4 +118,23 @@ function setCustomCacheControl(res, path) {
...
+    ]
+
+    for (const urlRegEx of iframeAllowedUrls)
```
Comment (blaisefonseca) [inline server.js:135]: For your particular case wouldn't just checking query string be better?

**Review by blaisefonseca — APPROVED** [review]

File: server.js (hunk 3 above)
Comment (randyhoulahan) [reply server.js:128]: Yes perhaps but then anyone can add the query string on and by pass blocked iframes for any route.  Better maybe just the route.  Neither route or QS is secret.

**Pull request merged**